cheatsheets/Authentication_Cheat_Sheet.md

```diff
@@ -85,24 +85,24 @@ For more information, see: [Client-authenticated TLS handshake](https://en.wikip
 
 ## Authentication and Error Messages
 
-Incorrectly implemented error messages in the case of authentication functionality can be used for the purposes of user ID and password enumeration. An application should respond (both HTTP and HTML) in a generic manner.
+Incorrectly implemented error messages in the case of authentication functionality can be used for the purposes of the user ID and password enumeration. An application should respond (both HTTP and HTML) in a generic manner.
 
 #### Authentication Responses
 
-Using any of the authentication mechanisms (login, password reset or password recovery) an application must respond with a generic error message regardless of whether:
+Using any of the authentication mechanisms (login, password reset or password recovery), an application must respond with a generic error message regardless of whether:
 * The user ID or password was incorrect.
 * The account does not exist.
 * The account is locked or disabled.
 
 The account registration feature should also be taken into consideration, and the same approach of generic error message can be applied regarding the case in which the user exists.
 
-The objective is to prevent the creation of a [discrepancy factor](https://cwe.mitre.org/data/definitions/204.html) allowing an attacker to mount a user enumeration action against the application.
+The objective is to prevent the creation of a [discrepancy factor](https://cwe.mitre.org/data/definitions/204.html), allowing an attacker to mount a user enumeration action against the application.
 
 It is interesting to note that the business logic itself can bring a discrepancy factor related to the processing time taken. Indeed, depending on the implementation, the processing time can be significantly different according to the case (success vs failure) allowing an attacker to mount a [time-based attack](https://en.wikipedia.org/wiki/Timing_attack) (delta of some seconds for example).
 
 Example using pseudo-code for a login feature:
 
-*First implementation using the "quick exit" approach*
+* the First implementation using the "quick exit" approach*
```

Comment:
This insertion is incorrect, it should just be `First`, matching the `Second` above the next one.